### PR TITLE
ci: new test case for https health check

### PR DESCRIPTION
**Pull request by Sn0rt**
### Description

https://github.com/apache/apisix/issues/9993

### Checklist

- [ ] I have explained the need for this PR and the problem it solves
- [ ] I have explained the changes or the new features added to this PR
- [ ] I have added tests corresponding to this change
- [ ] I have updated the documentation to reflect this change
- [ ] I have verified that this change is backward compatible (If not, please discuss on the [APISIX mailing list](https://github.com/apache/apisix/tree/master#community) first)

### CODE REVIEW

**Comment by shreemaan-abhishek** [comment]
also fixes: #10218

File: t/node/healthcheck-https.t [hunk 1]
```diff
@@ -0,0 +1,345 @@
...
+            assert(check_node_health(8767, "healthy"), "Port 8767 is not healthy")
+        }
+    }
+--- request
+GET /t
+--- error_code: 200
+
+
+
+=== TEST 1: https health check (one healthy node, one unhealthy node)
```
Comment (shreemaan-abhishek) [inline t/node/healthcheck-https.t:184]:
```suggestion
=== TEST 2: https health check (one healthy node, one unhealthy node)
```

File: t/node/healthcheck-https.t [hunk 2]
```diff
@@ -0,0 +1,345 @@
...
+--- request
+GET /t
+--- grep_error_log eval
+qr/\([^)]+\) unhealthy .* for '.*'/
+--- grep_error_log_out
+(upstream#/apisix/routes/1) unhealthy HTTP increment (1/1) for '127.0.0.1(127.0.0.1:8766)'
+
+
+
+=== TEST 2: https health check (two unhealthy nodes)
```
Comment (shreemaan-abhishek) [inline t/node/healthcheck-https.t:268]:
```suggestion
=== TEST 3: https health check (two unhealthy nodes)
```

**Review by shreemaan-abhishek — CHANGES_REQUESTED** [review]

File: t/node/healthcheck-https.t (hunk 1 above)
Comment (monkeyDluffy6017) [reply t/node/healthcheck-https.t:184]: please use `utils/reindex *.t` to reindex the file

Comment (Sn0rt) [reply t/node/healthcheck-https.t:184]: why CI not found this ?

image

File: t/node/healthcheck-https.t [hunk 3]
```diff
@@ -0,0 +1,345 @@
...
+                for _, node in ipairs(json_data.nodes) do
+                    if node.port == port and node.status == status then
+                        return true
+                    end
+                end
+                return false
+            end
+
+            assert(check_node_health(8765, "healthy"), "Port 8765 is not healthy")
+            assert(check_node_health(8766, "unhealthy"), "Port 8766 is not unhealthy")
```
Comment (monkeyDluffy6017) [inline t/node/healthcheck-https.t:256]:
```suggestion
            assert(check_node_health(8765, "healthy"), "Port 8765 is unhealthy")
            assert(check_node_health(8766, "unhealthy"), "Port 8766 is healthy")
```

File: t/node/healthcheck-https.t [hunk 4]
```diff
@@ -0,0 +1,345 @@
...
+                for _, node in ipairs(json_data.nodes) do
+                    if node.port == port and node.status == status then
+                        return true
+                    end
+                end
+                return false
+            end
+
+            assert(check_node_health(8766, "unhealthy"), "Port 8766 is not unhealthy")
+            assert(check_node_health(8768, "unhealthy"), "Port 8768 is not unhealthy")
```
Comment (monkeyDluffy6017) [inline t/node/healthcheck-https.t:340]:
```suggestion
            assert(check_node_health(8766, "unhealthy"), "Port 8766 is healthy")
            assert(check_node_health(8768, "unhealthy"), "Port 8768 is healthy")
```

File: t/node/healthcheck-https.t [hunk 5]
```diff
@@ -0,0 +1,345 @@
...
+                return
+            end
+
+            local http = require("resty.http")
+            for i = 1, 3 do
+                local httpc = http.new()
+                local uri = "http://127.0.0.1:" .. ngx.var.server_port .. "/ping"
+                local _, _ = httpc:request_uri(uri, {method = "GET", keepalive = false})
+                ngx.sleep(0.5)
+            end
```
[the remark this answers is not in the document]
Comment (Sn0rt) [reply t/node/healthcheck-https.t:237]: ok. I will update.

File: t/node/healthcheck-https.t [hunk 6]
```diff
@@ -0,0 +1,345 @@
...
+                for _, node in ipairs(json_data.nodes) do
+                    if node.port == port and node.status == status then
+                        return true
+                    end
+                end
+                return false
+            end
+
+            assert(check_node_health(8765, "healthy"), "Port 8765 is not healthy")
+            assert(check_node_health(8767, "healthy"), "Port 8767 is not healthy")
```
[the remark this answers is not in the document]
Comment (Sn0rt) [reply t/node/healthcheck-https.t:175]: this is assert, and refence `t/control/healthcheck.t:108`

**Comment by monkeyDluffy6017** [comment]
@shreemaan-abhishek please review

**Review by AlinsRan — APPROVED** [review]

**Review by luoluoyuyu — APPROVED** [review]

**Review by shreemaan-abhishek — APPROVED** [review]

**Pull request merged**